File: java/src/main/java/gov/doc/ntia/sigmf/Annotation.java
package gov.doc.ntia.sigmf;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;

public class Annotation implements Serializable {

  private static final long serialVersionUID = 1L;

  @JsonProperty(value = "core:sample_start", required = true)
  protected Long sampleStart;

  @JsonProperty(value = "core:sample_count", required = true)
  protected Long sampleCount;

  @JsonProperty(value = "core:generator", required = false)
  protected String generator;

  @JsonProperty(value = "core:comment", required = false)
  protected String comment;

  @JsonProperty(value = "core:freq_lower_edge", required = false)
  protected Double freqLowerEdge;

  @JsonProperty(value = "core:freq_upper_edge", required = false)
  protected Double freqUpperEdge;

  @JsonProperty(value = "uuid", required = false)
  protected String uuid;

  public Long getSampleStart() {
    return sampleStart;
  }

  public void setSampleStart(Long sampleStart) {
    this.sampleStart = sampleStart;
  }

  public Long getSampleCount() {
    return sampleCount;
  }

  public void setSampleCount(Long sampleCount) {
    this.sampleCount = sampleCount;
  }

  public String getGenerator() {
    return generator;
  }

  public void setGenerator(String generator) {
    this.generator = generator;
  }

  public String getComment() {
    return comment;
  }

  public void setComment(String comment) {
    this.comment = comment;
  }

  public Double getFreqLowerEdge() {
    return freqLowerEdge;
  }

  public void setFreqLowerEdge(Double freqLowerEdge) {
    this.freqLowerEdge = freqLowerEdge;
  }

  public Double getFreqUpperEdge() {
    return freqUpperEdge;
  }

  public void setFreqUpperEdge(Double freqUpperEdge) {
    this.freqUpperEdge = freqUpperEdge;
  }

  public String getUuid() {
    return uuid;
  }

  public void setUuid(String uuid) {
    this.uuid = uuid;
  }

}

File: Csharp/sigmf/sigmf/Annotation.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace sigmf
{
    [Serializable]
    public class Annotation
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("core:sample_start")]
        protected long sampleStart;

        [JsonProperty("core:sample_count")]
        protected long sampleCount;

        [JsonProperty("core:generator")]
        protected String generator;

        [JsonProperty("core:comment")]
        protected String comment;

        [JsonProperty("core:freq_lower_edge")]
        protected Double freqLowerEdge;

        [JsonProperty("core:freq_upper_edge")]
        protected Double freqUpperEdge;

        [JsonProperty("core:latitude")]
        protected Double latitude;

        [JsonProperty("core:longitude")]
        protected Double longitude;

        protected Dictionary<string, object> otherfields = new Dictionary<string,object>();

        public long SampleStart { get => sampleStart; set => sampleStart = value; }
        public long SampleCount { get => sampleCount; set => sampleCount = value; }
        public string Generator { get => generator; set => generator = value; }
        public string Comment { get => comment; set => comment = value; }
        public double FreqLowerEdge { get => freqLowerEdge; set => freqLowerEdge = value; }
        public double FreqUpperEdge { get => freqUpperEdge; set => freqUpperEdge = value; }
        public double Latitude { get => latitude; set => latitude = value; }
        public double Longitude { get => longitude; set => longitude = value; }

        public Dictionary<string, object> getOtherFields() {
            return otherfields;
        }
        public void add(string key, object value) {
            otherfields.Add(key, value);
        }
    }
}